Make chat filter only load once and general cleanup

jammer opened the pull request:
Removed the achievement "mute" feature that never activated because the particular achievements are not triggered.

Changed chat filter to load during PLAYER_ENTERING_WORLD event thus making it only load once instead of every join message.

Removed unnecessary checks from OnEvent as it is only registered for one event.

jammer committed:
Make chat filter only load once and general cleanup

Removed the achievement "mute" feature that never activated because the particular achievements are not triggered.

Changed chat filter to load during PLAYER_ENTERING_WORLD event thus making it only load once instead of every join message.

Removed unnecessary checks from OnEvent as it is only registered for one event.

jammer commented:
Current released version adds new chat filter every time someone joins the game. I'm unsure how much this will effect performance since the filter is rather simple, but seems like unnecessary load for long play sessions.

RaefWolfe reviewed (changes requested):
When I make these changes the noise from guild achievements still fires, even though the toast does not.

jammer committed:
Muting the achievement sound

Mistakenly assumed the alert triggered the achievement sound, which seems not to be the case so added back the muting of the sound.

jammer commented:
Sorry, I mistakenly assumed the sound was tied to the alert popup, but apparently not. I just added back the previous mute functionality into the alert function.
